gmso/core/topology.py

```diff
@@ -543,48 +543,88 @@ def update_connection_types(self):
-                    c.connection_type = self._improper_types[c.connection_type]
+            if not isinstance(connection_type_or_types, IndexedSet):
+                connection_type_or_types = [connection_type_or_types]
+            for connection_type in connection_type_or_types:
+                if connection_type not in self._connection_types:
+                    connection_type.topology = self
+                    self._connection_types[connection_type] = connection_type
+                    if isinstance(connection_type, BondType):
+                        self._bond_types[connection_type] = connection_type
+                        self._bond_types_idx[connection_type] = (
+                            len(self._bond_types) - 1
+                        )
+                    if isinstance(connection_type, AngleType):
+                        self._angle_types[connection_type] = connection_type
+                        self._angle_types_idx[connection_type] = (
+                            len(self._angle_types) - 1
+                        )
+                    if isinstance(connection_type, DihedralType):
+                        self._dihedral_types[connection_type] = connection_type
+                        self._dihedral_types_idx[connection_type] = (
+                            len(self._dihedral_types) - 1
+                        )
+                    if isinstance(connection_type, ImproperType):
+                        self._improper_types[
+                            connection_type
+                        ] = c.connection_type
+                        self._improper_types_idx[connection_type] = (
+                            len(self._improper_types) - 1
+                        )
+                elif connection_type in self.connection_types:
+                    if isinstance(connection_type, BondType):
+                        c.connection_type = self._bond_types[connection_type]
+                    if isinstance(connection_type, AngleType):
+                        c.connection_type = self._angle_types[connection_type]
+                    if isinstance(connection_type, DihedralType):
+                        if c.is_layered():
+                            if c.connection_types is None:
+                                c.connection_types = IndexedSet(
+                                    [self._dihedrals_types[connection_type]]
+                                )
+                            else:
+                                c.connection_types.add(
+                                    self._dihedral_types[connection_type]
+                                )
+                        else:
+                            c.connection_type = self._dihedral_types[
+                                connection_type
+                            ]
+                    if isinstance(connection_type, ImproperType):
+                        if c.is_layered():
+                            if c.connection_types is None:
```

Comment:
Based on the for loop [here](https://github.com/mosdef-hub/gmso/pull/569/files#diff-d27d58c8caf4bb1057b753cc0c2b4d3e1d65f5e1e4c4dd82ae4c99b525b9ba64R553-R559) We already check if the `connection_type` is `None` or not. And if it is, we `continue`. So these checks are not necessary

Reply:
Same for Dihedral above